web/js/codeworld_shared.js

```diff
@@ -532,14 +553,49 @@ function registerStandardHints(successFunc) {
       found[2] = [];
     }
 
-    const options = found[0].concat(found[1]).concat(found[2]);
-    for (const candidate of options) {
-      candidate.cost = substitutionCost(
+    const options = found[0].concat(found[1], found[2]);
+
+    options.forEach((candidate) => {
+      const { text } = candidate;
+      const originalTermCost = substitutionCost(
         token.string,
-        candidate.text,
+        text,
         term.length
       );
-    }
+
+      const mappedTerms =
```

Comment:
This is where the filtering by defining module should go, I think.  If the candidate has no defining module, or if its defining module doesn't match `replacementTerms`, then you should skip this rescoring logic.